Libraries/oneMKL/student_t_test/t_test.cpp

```diff
@@ -0,0 +1,201 @@
+// accepted, 0 - in case of NULL hypothesis should be rejected
+template <typename RealType>
+std::int32_t t_test(sycl::queue& queue, sycl::buffer<RealType, 1>& r1,
+                    std::int64_t n1, sycl::buffer<RealType, 1>& r2,
+                    std::int64_t n2) {
+  std::int32_t res = -1;
+
+  // Create buffers to be passed inside oneMKL stats functions
+  sycl::buffer<RealType, 1> mean1_buf(sycl::range{1});
+  sycl::buffer<RealType, 1> variance1_buf(sycl::range{1});
+  sycl::buffer<RealType, 1> mean2_buf(sycl::range{1});
+  sycl::buffer<RealType, 1> variance2_buf(sycl::range{1});
+  // Perform computations of mean and variance
+  auto dataset1 =
+      oneapi::mkl::stats::make_dataset<oneapi::mkl::stats::layout::row_major>(
+          1, n1, r1);
+  auto dataset2 =
+      oneapi::mkl::stats::make_dataset<oneapi::mkl::stats::layout::row_major>(
+          1, n2, r2);
+  oneapi::mkl::stats::mean(queue, dataset1, mean1_buf);
+  queue.wait_and_throw();
+  oneapi::mkl::stats::central_moment(queue, mean1_buf, dataset1, variance1_buf);
+  oneapi::mkl::stats::mean(queue, dataset2, mean2_buf);
+  queue.wait_and_throw();
+  oneapi::mkl::stats::central_moment(queue, mean2_buf, dataset2, variance2_buf);
+  queue.wait_and_throw();
+  // Create Host accessors and check the condition
+  auto mean1_acc = mean1_buf.template get_access<sycl::access::mode::read>();
+  auto variance1_acc =
+      variance1_buf.template get_access<sycl::access::mode::read>();
+  auto mean2_acc = mean2_buf.template get_access<sycl::access::mode::read>();
+  auto variance2_acc =
+      variance2_buf.template get_access<sycl::access::mode::read>();
+  bool almost_equal = (variance1_acc[0] < 2 * variance2_acc[0]) ||
+                      (variance2_acc[0] < 2 * variance1_acc[0]);
+  if (almost_equal) {
+    if ((sycl::abs(mean1_acc[0] - mean2_acc[0]) /
+         sycl::sqrt((static_cast<RealType>(1.0) / static_cast<RealType>(n1) +
+                     static_cast<RealType>(1.0) / static_cast<RealType>(n2)) *
+                    ((n1 - 1) * (n1 - 1) * variance1_acc[0] +
+                     (n2 - 1) * (n2 - 1) * variance2_acc[0]) /
+                    (n1 + n2 - 2))) < static_cast<RealType>(threshold)) {
+      res = 1;
+    } else {
+      res = 0;
+    }
+  } else {
+    if ((sycl::abs(mean1_acc[0] - mean2_acc[0]) /
+         sycl::sqrt((variance1_acc[0] + variance2_acc[0]))) <
+        static_cast<RealType>(threshold)) {
+      res = 1;
+    } else {
+      res = 0;
+    }
+  }
+  return res;
+}
+
+int main(int argc, char** argv) {
+  std::cout << "\nStudent's T-test Simulation\n";
+  std::cout << "Buffer Api\n";
+  std::cout << "-------------------------------------\n";
+
+  size_t n_points = n_samples;
+  fp_type mean = expected_mean;
+  fp_type std_dev = expected_std_dev;
+
+  if (argc >= 2) {
+    n_points = std::atol(argv[1]);
+    if (n_points == 0) {
+      n_points = n_samples;
+    }
+  }
+
+  if (argc >= 3) {
+    mean = std::atof(argv[2]);

```

Comment:
Do you need error checking on "mean" for bad command-line arguments?

Reply:
check was added